setup_configure.py

```diff
@@ -53,18 +53,30 @@ def validate_version(s):
 def mpi_enabled():
     return os.environ.get('HDF5_MPI') == "ON"
 
+def ros3_enabled():
+    hdf5 = os.environ.get('HDF5_DIR')
+
+    # Specified a prefix dir (e.g. '/usr/local')
+    if hdf5:
+        with open(op.join(hdf5, 'lib', 'libhdf5.settings')) as fp:
+            for line in fp.readlines():
+                if '(Read-Only) S3 VFD: yes' in line:
+                    return True
+    return False
```

Comment:
This seems a bit brittle. @aragilar has taken a different approach in #1759 - I haven't reviewed that in detail yet, but can I ask you to look at each other's PRs and figure out together what's the best way to detect optional build-time features? :+1:

Reply:
will coordinate with @aragilar . i'm happy to adopt whatever is the best way to detect build settings.

Reply:
I'm unclear whether `libhdf5.settings` is an explicit part of of the HDF5 install or just a helpful logfile, is it documented somewhere?

Reply:
if you do `h5cc -showconfig` it shows the exact contents of `libhdf5.settings` and if i remove it i get:

```
# h5cc -showconfig
cat: /usr/local/hdf5/lib/libhdf5.settings: No such file or directory
```

so i suspect this is an essential part of the library.

Reply:
@aragilar - any thoughts on the settings file?